src/v/cluster/cluster_discovery.cc

```diff
@@ -308,19 +308,18 @@ ss::future<> cluster_discovery::discover_founding_brokers() {
         }
         replies.emplace(seed_server.addr, cluster_bootstrap_info_reply{});
     }
-    co_await ss::parallel_for_each(
-      replies, ss::coroutine::lambda([this](auto& iter) -> ss::future<> {
-          auto& [addr, reply] = iter;
-          reply = co_await request_cluster_bootstrap_info_single(addr);
-          if (reply.cluster_uuid.has_value()) {
-              vlog(
-                clusterlog.info,
-                "Cluster presence detected in other seed servers: {}",
-                *reply.cluster_uuid);
-              _is_cluster_founder = false;
-              co_return;
-          }
-      }));
+    co_await ss::parallel_for_each(replies, [this](auto& entry) {
+        return request_cluster_bootstrap_info_single(entry.first)
```

Comment:
this is a use-after-free bug. to fix it, have `request_cluster_bootstrap_info_single` take it's arguments by value and not reference (it is a coroutine).